.github/workflows/arc-ci.yml

```diff
@@ -0,0 +1,59 @@
+name: Arc CI
+
+on: [push]
+
+# Some taken from
+# https://github.com/hiyainc-oss/trie-map/blob/5b892223e50854a4e0f9f86cd20c91efb5985126/.github/workflows/scala.yml
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v1    
+
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+
+    - name: Cache Ivy
+      id: cache-ivy
+      uses: actions/cache@v1.0.3
+      with:
+        path: ~/.ivy2
+        key: ${{ runner.os }}-ivy-${{ hashFiles('arc-frontend/**/*.sbt') }}
+
+    - name: Cache SBT
+      id: cache-sbt
+      uses: actions/cache@v1.0.3
+      with:
+        path: ~/.sbt
+        key: ${{ runner.os }}-sbt-${{ hashFiles('arc-frontend/**/*.sbt') }}
+
+    - name: Cache arc-frontend
+      id: cache-arc-frontend
+      uses: actions/cache@v1.0.3
+      with:
+        path: arc-frontend/target
+        key: ${{ runner.os }}-arc-frontend-${{ hashFiles('arc-frontend/**/*.sbt') }}-${{ hashFiles('arc-frontend/src/**/**') }}
+
+    - name: Cache arc-mlir
+      id: cache-arc-mlir
+      uses: actions/cache@v1.0.3
+      with:
+        path: arc-mlir/build
+        key: ${{ runner.os }}-arc-mlir-${{ hashFiles('arc-mlir/src/**/**') }}
+    
+    - name: Build and test arc-frontend
+      run: |
+        cd arc-frontend
+        sbt test assembly
+      
+    - name: Build and test arc-mlir
+      run: |
+        cd arc-mlir
```

Comment:
I cannot confirm it but I think GitHub will make sure everything is a clean sheet, and only what you explicitly cache should live across builds. It might be good either way to put all artefacts in the same top-level build directory. I will try and see if I can get the tests passing